[Flaky Test] aliases.js

```
  Aliases
    can be searched / sorted / paginated
      1) "before each" hook for "successfully"


  0 passing (1m)
  1 failing

  1) Aliases
       can be searched / sorted / paginated
         "before each" hook for "successfully":
     AssertionError: Timed out retrying after 60000ms: Expected to find content: 'Rows per page' but never did.

Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `Aliases`
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/integration/plugins/index-management-dashboards-plugin/aliases.js:132:8)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        7                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      6                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     1 minute, 7 seconds                                                              │
  │ Spec Ran:     plugins/index-management-dashboards-plugin/aliases.js                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

```

https://github.com/opensearch-project/index-management-dashboards-plugin/actions/runs/6117128934/job/16603304492

https://github.com/opensearch-project/index-management-dashboards-plugin/actions/runs/6714280704/job/18247266546?pr=909

https://github.com/opensearch-project/index-management-dashboards-plugin/actions/runs/6778757305/job/18425250051?pr=918

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -25,24 +25,6 @@ describe("Aliases", () => {
     cy.addIndexAlias(`${SAMPLE_ALIAS_PREFIX}-0`, `${SAMPLE_INDEX_PREFIX}-*`);
   });
 
-  Cypress.Commands.add("waitForAliasWithRetry", (alias, retryCount = 3, timeout = 240000) => {
-    const tryWait = (currentAttempt) => {
-      cy.wait(alias, { timeout: timeout }).then(
-        () => true, // If successful, do nothing
-        (error) => {
-          if (currentAttempt < retryCount) {
-            cy.log(`Retry attempt ${currentAttempt + 1} for ${alias}`);
-            tryWait(currentAttempt + 1); // Retry
-          } else {
-            throw error; // If retries are exhausted, throw the error
-          }
-        }
-      );
-    };
-
-    tryWait(0); // Initial call to tryWait
-  });
-
   beforeEach(() => {
     // Intercept the specific POST request
     cy.intercept("POST", "/api/ism/apiCaller", (req) => {
@@ -58,8 +40,8 @@ describe("Aliases", () => {
     // eslint-disable-next-line cypress/no-unnecessary-waiting
     cy.wait(120000);
 
-    // Use the custom command for waiting with retries
-    cy.waitForAliasWithRetry("@apiCaller");
+    // Wait for the API call to complete
+    cy.wait("@apiCaller", { timeout: 240000 });
 
     // Common text to wait for to confirm page loaded, give up to 120 seconds for initial load
     cy.contains("Rows per page", { timeout: 120000 }).should("be.visible");

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -33,12 +33,12 @@ describe("Aliases", () => {
       }
     });
 
-    // Wait for 10 seconds for OSD to start.
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(10000);
-
     // Visit ISM OSD
     cy.visit(`${BASE_PATH}/app/${IM_PLUGIN_NAME}#/aliases`);
+
+    // Wait for 30 seconds for OSD to start.
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(30000);
 
     // Wait for the API call to complete
     cy.wait("@apiCaller", { timeout: 120000 });

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -38,10 +38,17 @@ describe("Aliases", () => {
 
     // Wait for 120 seconds for OSD to start.
     // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(120000);
+    // cy.wait(120000);
+
+    const startTime = new Date().getTime();
 
     // Wait for the API call to complete
-    cy.wait("@apiCaller", { timeout: 240000 });
+    cy.wait("@apiCaller", { timeout: 240000 }).then(() => {
+      // Log the calculated duration
+      const endTime = new Date().getTime();
+      const duration = endTime - startTime; // Duration in milliseconds
+      cy.log(`@apiCaller completed in ${duration} milliseconds`);
+    });
 
     // Common text to wait for to confirm page loaded, give up to 120 seconds for initial load
     cy.contains("Rows per page", { timeout: 120000 }).should("be.visible");

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -33,6 +33,10 @@ describe("Aliases", () => {
       }
     });
 
+    // Wait for 10 seconds for OSD to start.
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(10000);
+
     // Visit ISM OSD
     cy.visit(`${BASE_PATH}/app/${IM_PLUGIN_NAME}#/aliases`);
 

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -25,6 +25,24 @@ describe("Aliases", () => {
     cy.addIndexAlias(`${SAMPLE_ALIAS_PREFIX}-0`, `${SAMPLE_INDEX_PREFIX}-*`);
   });
 
+  Cypress.Commands.add("waitForAliasWithRetry", (alias, retryCount = 3, timeout = 240000) => {
+    const tryWait = (currentAttempt) => {
+      cy.wait(alias, { timeout: timeout }).then(
+        () => true, // If successful, do nothing
+        (error) => {
+          if (currentAttempt < retryCount) {
+            cy.log(`Retry attempt ${currentAttempt + 1} for ${alias}`);
+            tryWait(currentAttempt + 1); // Retry
+          } else {
+            throw error; // If retries are exhausted, throw the error
+          }
+        }
+      );
+    };
+
+    tryWait(0); // Initial call to tryWait
+  });
+
   beforeEach(() => {
     // Intercept the specific POST request
     cy.intercept("POST", "/api/ism/apiCaller", (req) => {
@@ -36,15 +54,15 @@ describe("Aliases", () => {
     // Visit ISM OSD
     cy.visit(`${BASE_PATH}/app/${IM_PLUGIN_NAME}#/aliases`);
 
-    // Wait for 30 seconds for OSD to start.
+    // Wait for 120 seconds for OSD to start.
     // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(30000);
+    cy.wait(120000);
 
-    // Wait for the API call to complete
-    cy.wait("@apiCaller", { timeout: 120000 });
+    // Use the custom command for waiting with retries
+    cy.waitForAliasWithRetry("@apiCaller");
 
-    // Common text to wait for to confirm page loaded, give up to 60 seconds for initial load
-    cy.contains("Rows per page", { timeout: 60000 }).should("be.visible");
+    // Common text to wait for to confirm page loaded, give up to 120 seconds for initial load
+    cy.contains("Rows per page", { timeout: 120000 }).should("be.visible");
   });
 
   describe("can be searched / sorted / paginated", () => {

--- a/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
+++ b/cypress/integration/plugins/index-management-dashboards-plugin/aliases.js
@@ -26,11 +26,21 @@ describe("Aliases", () => {
   });
 
   beforeEach(() => {
+    // Intercept the specific POST request
+    cy.intercept("POST", "/api/ism/apiCaller", (req) => {
+      if (req.body.data && req.body.data.name === "**" && req.body.data.s === "alias:desc" && req.body.endpoint === "cat.aliases") {
+        req.alias = "apiCaller"; // Assign an alias directly if the condition is met
+      }
+    });
+
     // Visit ISM OSD
     cy.visit(`${BASE_PATH}/app/${IM_PLUGIN_NAME}#/aliases`);
 
+    // Wait for the API call to complete
+    cy.wait("@apiCaller", { timeout: 120000 });
+
     // Common text to wait for to confirm page loaded, give up to 60 seconds for initial load
-    cy.contains("Rows per page", { timeout: 60000 });
+    cy.contains("Rows per page", { timeout: 60000 }).should("be.visible");
   });
 
   describe("can be searched / sorted / paginated", () => {

